src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8065,8 +8077,95 @@ namespace WaterThermalTanks {
+                        dT_max = min(dT_max, Tank.SetPointTemp2 - Tank.Node(Tank.HeaterNode2).Temp);
+                    }
+                }
+
+                // Make adjustments to A and B to account for heaters being on or off now
+                if (Tank.HeaterOn1 and !PrevHeaterOn1) {
+                    // If heater 1 is on now and wasn't before add the heat rate to the B term
+                    B[Tank.HeaterNode1 - 1] += Qheater1 / (Tank.Node(Tank.HeaterNode1).Mass * Cp);
+                } else if (!Tank.HeaterOn1 and PrevHeaterOn1) {
+                    // If heater 1 is off now and was on before, remove the heat rate from the B term
+                    B[Tank.HeaterNode1 - 1] -= Tank.MaxCapacity / (Tank.Node(Tank.HeaterNode1).Mass * Cp);
+                }
+                if (Tank.HeaterOn2 and !PrevHeaterOn2) {
+                    // If heater 2 is on now and wasn't before add the heat rate to the B term
+                    B[Tank.HeaterNode2 - 1] += Qheater2 / (Tank.Node(Tank.HeaterNode2).Mass * Cp);
+                } else if (!Tank.HeaterOn2 and PrevHeaterOn2) {
+                    // If heater 1 is off now and was on before, remove the heat rate from the B term
+                    B[Tank.HeaterNode2 - 1] -= Tank.MaxCapacity / (Tank.Node(Tank.HeaterNode2).Mass * Cp);
+                }
+
+                if ((Tank.HeaterOn1 || Tank.HeaterOn2) and !(PrevHeaterOn1 || PrevHeaterOn2)) {
+                    // Remove off cycle loads
+                    // Apply on cycle loads
+                    for (int i = 0; i < nTankNodes; i++) {
+                        auto &node(Tank.Node[i]);
+                        Real64 NodeCapacitance = node.Mass * Cp;
+                        A[i] += (node.OffCycLossCoeff - node.OnCycLossCoeff) / NodeCapacitance;
+                        B[i] += (- node.OffCycParaLoad + node.OnCycParaLoad + (node.OnCycLossCoeff - node.OffCycLossCoeff) * Tank.AmbientTemp) / NodeCapacitance;
+                    }
+                } else if (!(Tank.HeaterOn1 || Tank.HeaterOn2) and (PrevHeaterOn1 || PrevHeaterOn2)) {
+                    // Remove on cycle loads
+                    // Apply off cycle loads
+                    for (int i = 0; i < nTankNodes; i++) {
+                        auto &node(Tank.Node[i]);
+                        Real64 NodeCapacitance = node.Mass * Cp;
+                        A[i] -= (node.OffCycLossCoeff - node.OnCycLossCoeff) / NodeCapacitance;
+                        B[i] -= (- node.OffCycParaLoad + node.OnCycParaLoad + (node.OnCycLossCoeff - node.OffCycLossCoeff) * Tank.AmbientTemp) / NodeCapacitance;
+                    }
+                }
+
+                // Set the sub timestep (dt)
+                dt = TimeRemaining;
+                for (int i = 0; i < nTankNodes; ++i) {
+                    const Real64 Denominator = fabs(A[i] * Tavg[i] + B[i]);
+                    if (Denominator != 0.0)
+                        dt = min(dt, dT_max / Denominator);
```

Comment:
If nTankNodes > 1, then how is dt adjusted? It appears the nth node will set dt. Shouldn't there be a += here and then a divide by nTankNodes after the for loop?

Reply:
`dt` is determined by solving

![image](https://user-images.githubusercontent.com/5325034/63442401-47761880-c3f0-11e9-94e0-1efd9bf1f0d7.png)

for dt. We determine the maximum allowable temperature change, `dT_max`,  for any node above. In these lines we estimate the time it would take for each node to experience that much of a temperature change. Then I take the minimum of those so that we only go as far forward as the fastest changing node temperature, which isn't necessarily `n`. That's what the `min` here does. It's equivalent to writing:

```c++
if (dT_max / Denominator < dt) 
    dt = dT_max / Denominator
```

That way we're calculating the min dt of all the nodes.

Reply:
OK, I missed the "min(dt," part of it. This looks good then with using the smallest dt.

min(dt, dT_max / Denominator);